### PR TITLE
Remove dependency to KeyVal launcher when it is not needed

### PR DESCRIPTION
**Pull request by mariomac**
The GRPC instrumentors require to expose a writable memory map, so the [KeyVal launcher](https://github.com/keyval-dev/launcher) is needed.

However, for HTTP instrumentation don't require it, so the auto-instrumenter process should not fail if that mmap is not available. This will simplify the deployment in many scenarios.

The PR includes an example deployment. However it won't still provide metrics because the HTTP instrumenter needs to be fixed (will do it in another PR).

This should help with issue #17

### CODE REVIEW

File: pkg/process/analyze.go [hunk 1]
```diff
@@ -86,10 +85,14 @@ func (a *processAnalyzer) findKeyvalMmap(pid int) (uintptr, uintptr) {
...
 			log.Logger.Info("found addr of keyval map", "addr", m.StartAddr)
-			return m.StartAddr, m.EndAddr
+			return &AllocationDetails{
+				StartAddr: uint64(m.StartAddr),
+				EndAddr:   uint64(m.EndAddr),
+			}
 		}
 	}
-	panic(errors.New("cant find keyval map"))
+	log.Logger.V(1).Info("did not find any address for keyval map. Some services couldn't be instrumented")
```
Comment (edeNFed) [inline pkg/process/analyze.go:94]: I think a better message would be something like `could not allocate remote memory, automatic context propagation will be disabled`

**Review by edeNFed — APPROVED** [review]
lgtm

File: deployments/grafana-demo/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# How to deploy
```
Comment (MrAlias) [inline deployments/grafana-demo/README.md:1]: Can we split this demo example to its own PR?

Comment (mariomac) [reply deployments/grafana-demo/README.md]: Done! Here is the PR: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/37

**Review by MrAlias — APPROVED** [review]
